### PR TITLE
Sidebar files

### PR DESCRIPTION
**Pull request by siddharthkp**
Built on top of https://github.com/codesandbox/codesandbox-client/pull/3399

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3405/commits/7712f4020cfd2566cbd902e3f27dd0d37473500e failed.

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+export class Dependency extends React.PureComponent<Props, State> {
+  state: State = {
+    hovering: false,
+    version: null,
+    open: false,
+    versions: [],
+  };
+
+  setVersionsForLatestPkg(pkg: string) {
+    const that = this;
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx]: wat?

Comment (SaraVieira) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx]: I know this is copied btw

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+              right: 0,
+              width: '160px',
+            })}
+          >
+            {hovering && (
+              <>
+                <Select
+                  css={{ width: '80px' }}
+                  defaultValue={versions.find(
+                    v => v === dependencies[dependency]
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx]: can you rename this v to version, it was probably me :(

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx (hunk 1 above)
Comment (siddharthkp) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx]: same 😅 didn't even read it tbh

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+}
+
+interface State {
+  hovering: boolean;
+  version: null | string;
+  open: boolean;
+  versions: string[];
+}
+
+export class Dependency extends React.PureComponent<Props, State> {
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx]: Is there a way you can make this not a class?

Comment (siddharthkp) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx]: can refactor the copied code, didn't make an attempt 👼 8 j

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/ExternalResources/index.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                <CrossIcon />
+              </Button>
+            </ListAction>
+          ))}
+        </List>
+
+        <form
+          key={otherResources.length}
+          onSubmit={event => {
+            event.preventDefault();
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/ExternalResources/index.tsx]: Can you please move this to a function? PLs? 🙏

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/ExternalResources/index.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            <Button variant="secondary" type="submit">
+              Add resource
+            </Button>
+          </SidebarRow>
+        </form>
+
+        <form
+          key={fonts.length}
+          onSubmit={event => {
+            event.preventDefault();
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/ExternalResources/index.tsx]: Same :(

**Comment by SaraVieira** [comment]
Soo, it looks good and I am gonna approve it and then we should work on improving this code but I can do that after merge

I like refactoring 😂

**Pull request merged**